### PR TITLE
Unfixamo2

### PR DESCRIPTION
**Pull request by bovine3dom**
Comply with AMO request to instead remove pref from user.js.

Do not merge until AMO has confirmed if they want us to remove `resistFingerprinting` too and this PR matches up with their desire. I really don't want an `unfixamo3` :(

### CODE REVIEW

File: src/lib/native.ts [hunk 1]
```diff
@@ -758,33 +758,53 @@ export async function writePref(name: string, value: any) {
...
+        if (localStorage.unfixedamo2 === "true") {
             // unfixamo already ran for the tridactyl instance in this profile
             return
         }
 
         const profile = (await getProfileDir()) + "/"
         const userjs = await loadPrefs(profile + "user.js")
         const tridactylPref = "tridactyl.unfixedamo"
+        const restricted = "extensions.webextensions.restrictedDomains"
         if (userjs[tridactylPref] === "true") {
```
Comment (glacambre) [inline src/lib/native.ts]: Can't this just be `if (userjs[tridactylPref2] === "true") return;`? And then have line 798 check if `userjs[restricted] === "" || userjs[restricted] === restrictedDomains`.

File: src/lib/native.ts [hunk 2]
```diff
@@ -758,33 +758,53 @@ export async function writePref(name: string, value: any) {
...
+        substr = text.substring(prefPos, prefPos + prefEnd)
+        write(file, text.replace(substr, ``))
+    }
+}
+
 /** Obey Mozilla's orders https://github.com/tridactyl/tridactyl/issues/1800 */
 export async function unfixamo() {
     try {
-        if (localStorage.unfixedamo === "true") {
+        if (localStorage.unfixedamo2 === "true") {
```
[the remark this answers is not in the document]
Comment (bovine3dom) [reply src/lib/native.ts]: Isn't this a good thing? It means it won't run repeatedly. I don't quite understand what you mean. Are you suggesting we shouldn't 'fix' it for beta users?

[the remark this answers is not in the document]
Comment (glacambre) [reply src/lib/native.ts]: >Are you suggesting we shouldn't 'fix' it for beta users?

Yeah, I'm not sure it's a good idea but I was thinking that Mozilla probably only cares about stable users and that we thus shouldn't make the code more complicated than it needs to be. But I could be wrong.

[the remark this answers is not in the document]
Comment (bovine3dom) [reply src/lib/native.ts]: I think it isn't too complicated. I didn't like leaving the restrictedDomains line there anyway.

**Comment by bovine3dom** [comment]
I don't want to ping everyone on #1800 with a non-update so I'll do it here: we're waiting for the AMO to get back to us on whether they wanted us to remove the resistFingerprinting setting too. I just sent an email chasing it up.

File: src/lib/native.ts [hunk 3]
```diff
@@ -758,36 +758,57 @@ export async function writePref(name: string, value: any) {
...
         const profile = (await getProfileDir()) + "/"
         const userjs = await loadPrefs(profile + "user.js")
         const tridactylPref = "tridactyl.unfixedamo"
+        const tridactylPref2 = "tridactyl.unfixedamo_removed"
+        const restricted = "extensions.webextensions.restrictedDomains"
+        if (userjs[tridactylPref2] === "true") return;
         if (userjs[tridactylPref] === "true") {
-            // unfixamo already ran for this firefox profile
+            // unfixamo already ran for this Firefox profile in the Beta
+            await removePref(restricted)
```
Comment (bovine3dom) [inline src/lib/native.ts]: Reviewers also want block_mozAddonManager removed.

**Pull request merged**